src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -965,6 +954,42 @@ public function offsetGet($option)
     }
 
     /**
+     * Verify value is of the allowed type. Recursive method to support
+     * typed array notation like ClassName[], or scalar arrays (int[]).
+     *
+     * @param string $type  the required allowedType string
+     * @param mixed  $value the value
+     *
+     * @return bool Whether or not $value if of the allowed type
+     */
+    private function verifyAllowedType($type, $value)
+    {
+        $valid = false;
+
+        $type = isset(self::$typeAliases[$type]) ? self::$typeAliases[$type] : $type;
+
+        if (substr($type, -2) === '[]') {
+            if (is_array($value)) {
+                $subType = substr($type, 0, -2);
+                foreach ($value as $v) {
+                    $valid = $this->verifyAllowedType($subType, $v);
+                    if (!$valid) {
+                        break;
```

Comment:
just `` return false`` here, and ``return true`` after the loop. We tend to use early returns rather than making lots of elseif calls and storing the return value in a variable until the end of the method (it makes it much more obvious that we found the return value)